metal_header/sifive_pwm0.c++

```diff
@@ -0,0 +1,350 @@
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("interrupt_lines", "int",
+                               "struct metal_pwm *pwm", "int idx");
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("clock", "struct metal_clock *",
+                               "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("pinmux", "struct __metal_driver_sifive_gpio0 *",
+                               "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("pinmux_output_selector", "unsigned long",
+                               "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("pinmux_source_selector", "unsigned long",
+                               "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+
+      func = create_inline_dec("compare_width", "int", "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+
+      func =
+          create_inline_dec("comparator_count", "int", "struct metal_pwm *pwm");
+      extern_inlines.push_back(func);
+    }
+    count++;
+  });
+  os << "\n";
+  os << "/* --------------------- sifive_pwm0 ------------ */\n";
+  while (!extern_inlines.empty()) {
+    func = extern_inlines.front();
+    extern_inlines.pop_front();
+    emit_inline_dec(func, "sifive_pwm0");
+    delete func;
+  }
+  os << "\n";
+}
+
+void sifive_pwm0::define_inlines() {
+
+  Inline *control_base_func;
+  Inline *control_size_func;
+  Inline *num_interrupts_func;
+  Inline *clock_func;
+  Inline *pinmux_func;
+  Inline *pinmux_output_selector_func;
+  Inline *pinmux_source_selector_func;
+  Inline *interrupt_parent_func;
+  Inline *interrupt_line_func;
+  Inline *compare_width_func;
+  Inline *comparator_count_func;
+  int count = 0;
+
+  dtb.match(std::regex(compat_string), [&](node n) {
+    /* Clock driving the pwm peripheral */
+    std::string clock_value = "NULL";
+    n.maybe_tuple("clocks", tuple_t<node>(), [&]() {},
+                  [&](node m) {
+                    clock_value = "(struct metal_clock *)&__metal_dt_" +
+                                  m.handle() + ".clock";
+                  });
+
+    /* Pinmux */
+    std::string pinmux_value = "NULL";
+    uint32_t pinmux_dest = 0;
+    uint32_t pinmux_source = 0;
+    n.maybe_tuple("pinmux", tuple_t<node, uint32_t, uint32_t>(), [&]() {},
+                  [&](node m, uint32_t dest, uint32_t source) {
+                    pinmux_value =
+                        "(struct __metal_driver_sifive_gpio0 *)&__metal_dt_" +
+                        m.handle();
+                    pinmux_dest = dest;
+                    pinmux_source = source;
+                  });
+
+    /* Comparator count and width */
+    uint32_t cmpwidth = 0;
+    uint32_t ncmp = 0;
+    n.maybe_tuple("ncmp_cmpwidth", tuple_t<uint32_t, uint32_t>(), [&]() {},
```

Comment:
The string should be "sifive,comparator-widthbits". Also check for existence, if not available default to 16.
Similarly, check for existence if "sifive,ncomparators". If not available, default to 1.

Reply:
String updated, existence check added. Default values are set.